modules/evm/src/runner/stack.rs

```diff
@@ -69,7 +69,12 @@ impl<T: Config> Runner<T> {
 		F: FnOnce(&mut StackExecutor<'config, SubstrateStackState<'_, 'config, T>>) -> (ExitReason, R),
 	{
 		let gas_price = U256::one();
-		let vicinity = Vicinity { gas_price, origin };
+		let vicinity = Vicinity {
+			gas_price,
+			origin,
+			block_coinbase: Pallet::<T>::find_author(),
```

Comment:
And `block_difficulty` and `block_gas_limit `. If it is not used, it is better not to add it, not to maintain it

Reply:
we could also feature gate the new fields behind `evm-tests` as is done elsewhere, if it is needed to pass the evm-tests module

Reply:
I saw it used in evm-tests.  Maybe use `Option<H160>` for `block_coinbase` is better.

Reply:
> I don't see the use of adding `block_coinbase`, instead it adds a query per transaction

https://github.com/AcalaNetwork/Acala/pull/1792/files#diff-80b1f4d5e20f959107caa50e5409d8d2b0cb931a6eaef7b69edccc813f39aa87L460

it is used. I just moved it to vicinity because it's easy for testing as well. Also `find_author()` wont be called each time evm calls coinbase

Reply:
regarding `block_difficulty` and `block_gas_limit` they will just be zeros, nothing to maintain

Reply:
> Also find_author() wont be called each time evm calls coinbase

I mean any tx(e.g. erc20 transfer) will call `find_author()`. 

`Opcode::COINBASE => system::coinbase(state, handler)`
Only `opcode` will call `block_coinbase`. So use Option is better.

Reply:
Yes

Reply:
I see

Reply:
I guess we can do the same for `block_limit` and `difficulty` unwrap_or_default()